### PR TITLE
Handle case where applocker test script fails to delete

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

One reported case where during applocker policy check, the test script that is created cannot be deleted because something else (malware scanner?) has a lock on the file.  This results in an exception being thrown in finally block which ultimately causes PowerShell to crash due to the unhandled exception.  Fix is to wrap the deletion in the finally block with try..catch and ignore any exceptions and leave the temp file.

TFS:20156282

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by PaulHigin — APPROVED** [review]

File: src/System.Management.Automation/security/wldpNativeMethods.cs [hunk 1]
```diff
@@ -277,9 +277,29 @@ private static SystemEnforcementMode GetAppLockerPolicy(string path, SafeHandle
...
                     finally
                     {
-                        if (IO.File.Exists(testPathScript)) { IO.File.Delete(testPathScript); }
+                        if (IO.File.Exists(testPathScript))
+                        {
+                            try
+                            {
+                                IO.File.Delete(testPathScript);
+                            }
+                            catch
```
Comment (anmenaga) [inline src/System.Management.Automation/security/wldpNativeMethods.cs]: Maybe instead of catching everything, catch only IOException?
Same for below.

Comment (iSazonov) [reply src/System.Management.Automation/security/wldpNativeMethods.cs]: And I'd move this dup code in helper method `TryDeleteFile()`.

**Comment by iSazonov** [comment]
If we can not delete a file is it secure to leave it on file system?

File: src/System.Management.Automation/security/wldpNativeMethods.cs (hunk 1 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/security/wldpNativeMethods.cs]: Will update the code based on suggestions

**Comment by SteveL-MSFT** [comment]
@iSazonov yes, the test script is harmless and doesn't expose anything, it just validates whether scripts can be run or not.  It is not ideal to leave the file in the temp folder, but worse is to prevent PowerShell from starting simply because something else decided to lock that file.

**Comment by SteveL-MSFT** [comment]
@PaulHigin can you take another look?  Made some changes based on feedback

**Review by iSazonov — APPROVED** [review]

**Review by PaulHigin — APPROVED** [review]

**Pull request merged**